src/decorators.ts

```diff
@@ -1,15 +1,31 @@
-import {Container} from "./Container";
+
+import {Container, ServiceDescriptor} from "./Container";
+
 
 /**
  * Marks class as a service that can be injected using container.
  *
- * @param name Optional service name can be specified. If service name is specified then this service can only be
- *              retrieved by a service name. If no service name is specified then service can be retrieved by its type
+ * @param descriptor
  */
-export function Service(name?: string) {
+export function Service(descriptor: ServiceDescriptor = {}) {
```

Comment:
The most important always is public api. Which one do you think is better clear and straightforward for user `@Service(descriptor: ServiceDescriptor = {}) ` or `Service(descriptor?: ServiceDescriptor)`.  Implementation is just one extra check.

Also, is `ServiceDescriptor` isn't best name here. Better to do `@Service(options?: ServiceOptions)`. And better to left this object dummy and create a new object `ServiceMetadata` and copy all options into it. I'm following this pattern in all libs. If you are confused and don't get the idea just left it to me, I'll refactor once I merge these changes.